todo-app/src/TodoApp.js

```diff
@@ -1,17 +1,64 @@
 import React, { Component } from 'react';
 import TodoList from './TodoList';
+import todoItems from './todo.json';
 
 export default class TodoApp extends Component {
 
+  state = {
+    commentText: '',
+    todos: todoItems
+  };
+
+
+  onAddClick = (event) => {
+    event.preventDefault();
+    var text = this.state.commentText;
+    console.log("here is :  ", text);
+
+    const ids = this.state.todos.map(todos => todos.id);
+    
+    const addTodo = {
+      "id": ids.length === 0 ? 1 : Math.max(...ids) + 1,
+      "description": text,
+      "deadline": "2018-01-11",
+      "done": false
+    }
+
+    var updatedTodos = [addTodo, ...this.state.todos]
+
+    this.setState({
+      commentText: '',
+      todos: updatedTodos
+    });
+  };
+
+  onAddChange = event => {
+    var text = event.target.value;
+    this.setState({
+      commentText: text
+    });
+  };
+
+    
+
   render() {
     return (
       <div className="TodoApp">
         <h2>TODO LIST</h2>
+        <div>
+          <div>
+            <textarea
+            className='TextField'
+            value={this.state.commentText}
+            onChange={this.onAddChange}
+            />
+          </div>
+          <div>
+            <button className="add-button" onClick={this.onAddClick}> Add</button>
+          </div>  
+        </div>
         <TodoList/>
```

Comment:
Hint...

Reply:
Unfortunately, I couldn't get it. @isaachinman

Reply:
I could make it, if I could set the state in TodoList. 
I could't figure out how to carry the state down to the tree.

Reply:
@pesh12 Yes, you have the right idea.

The problem is that in both `TodoApp.js` and `TodoList.js` you are loading data from the json file into state. This means that even when `TodoApp` changes its todos, `TodoList` doesn't care as it has its own version of the list in internal state.

You need to pass `TodoApp`'s state into `TodoList` like this:

    <TodoList todos={this.state.todos} />

And then wherever you are referencing `this.state.todos` inside `TodoList`, simply change to `this.props.todos`.

Let me know how it goes...

Reply:
I made a function in TodoList, to handle the "statechange"

It did not change the state, but changed the props (adds the value in TodoList). 
But I put a console.log in the function, it never works.
I think I am really confused :))
We may sit on Sunday and see the problem together.